test/testdrive/expected_group_size_tuning.td

```diff
@@ -0,0 +1,280 @@
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Start from a TPC-H load generator source with small data.
+> CREATE SOURCE lgtpch FROM LOAD GENERATOR TPCH (SCALE FACTOR 0.0001, TICK INTERVAL 0.1) FOR ALL TABLES WITH (SIZE = '1');
+
+# Create a set of materialized views for testing based on the TPC-H schema.
+> CREATE MATERIALIZED VIEW lineitem_by_orderkey AS
+  SELECT l1.l_orderkey,
+        MAX(l1.l_extendedprice),
+        (SELECT l2.l_quantity FROM lineitem l2
+        WHERE l2.l_orderkey = l1.l_orderkey
+        ORDER BY l2.l_extendedprice DESC
+        LIMIT 1)
+  FROM lineitem l1
+  GROUP BY l1.l_orderkey;
+
+> CREATE MATERIALIZED VIEW lineitem_by_partsuppkey AS
+  SELECT l1.l_partkey,
+        l_suppkey,
+        MAX(l1.l_extendedprice),
+        (SELECT l2.l_quantity FROM lineitem l2
+        WHERE l2.l_partkey = l1.l_partkey
+          AND l2.l_suppkey = l1.l_suppkey
+        ORDER BY l2.l_extendedprice DESC
+        LIMIT 1)
+  FROM lineitem l1
+  GROUP BY l1.l_partkey, l_suppkey;
+
+> CREATE MATERIALIZED VIEW lineitem_by_partkey AS
+  SELECT l1.l_partkey,
+        MAX(l1.l_extendedprice),
+        (SELECT l2.l_quantity FROM lineitem l2
+        WHERE l2.l_partkey = l1.l_partkey
+        ORDER BY l2.l_extendedprice DESC
+        LIMIT 1)
+  FROM lineitem l1
+  GROUP BY l1.l_partkey;
+
+> CREATE MATERIALIZED VIEW lineitem_by_suppkey AS
+  SELECT l1.l_suppkey,
+        MAX(l1.l_extendedprice),
+        (SELECT l2.l_quantity FROM lineitem l2
+        WHERE l2.l_suppkey = l1.l_suppkey
+        ORDER BY l2.l_extendedprice DESC
+        LIMIT 1)
+  FROM lineitem l1
+  GROUP BY l1.l_suppkey;
+
+# Check that we know the values of the perfect hints for TPC-H.
+> SELECT pow(16, ceil(log(16, MAX(group_size)))) - 1 AS perfect_hint
+  FROM (
+        SELECT l_orderkey, COUNT(*) AS group_size
+        FROM lineitem
+        GROUP BY l_orderkey
+  );
+15
+
+> SELECT pow(16, ceil(log(16, MAX(group_size)))) - 1 perfect_hint
+  FROM (
+        SELECT l_partkey, l_suppkey, COUNT(*) AS group_size
+        FROM lineitem
+        GROUP BY l_partkey, l_suppkey
+  );
+255
+
+> SELECT pow(16, ceil(log(16, MAX(group_size)))) - 1 perfect_hint
+  FROM (
+        SELECT l_partkey, COUNT(*) AS group_size
+        FROM lineitem
+        GROUP BY l_partkey
+  );
+255
+
+> SELECT pow(16, ceil(log(16, MAX(group_size)))) - 1 perfect_hint
+  FROM (
+        SELECT l_suppkey, COUNT(*) AS group_size
+        FROM lineitem
+        GROUP BY l_suppkey
+  );
+4095
+
+> WITH operators AS (
+  SELECT
+      dod.dataflow_id,
+      dor.id AS region_id,
+      dod.id,
+      ars.records
+  FROM
+      mz_internal.mz_dataflow_operator_dataflows dod
+      JOIN mz_internal.mz_dataflow_addresses doa
+          ON dod.id = doa.id
+      JOIN mz_internal.mz_dataflow_addresses dra
+          ON CAST(dra.address AS text) = CAST(doa.address[:list_length(doa.address) - 1] as text)
```

Comment:
Comparing lists directly without casting them to text is not possible?

Reply:
It is, good catch!